server/servers/api/src/test/scala/com/prisma/api/mutations/nonEmbedded/VeryManyMutationsSpec.scala

```diff
@@ -88,6 +76,87 @@ class VeryManyMutationsSpec extends FlatSpec with Matchers with ApiSpecBase {
          |  ]}
          |}) {int}}"""
 
-    server.query(query, project)
+      server.query(query, project)
+    }
+
+    for (int <- 1 to 1000) {
+      createTop(int)
+    }
+
+    val update = server.query("""mutation {updateManyMiddles(where: { int_gt: 100 } data:{int: 500}){count}}""", project)
+    update.pathAsLong("data.updateManyMiddles.count") should equal(36291)
+
+    val result = server.query("""mutation {deleteManyMiddles(where: { int_gt: 100 }){count}}""", project)
+    result.pathAsLong("data.deleteManyMiddles.count") should equal(36291)
+  }
+
+  "A cascading delete" should "not hit the parameter limit" in {
+
+    val project: Project = SchemaDsl.fromString() {
+      """
+        |type Top {
+        |   id: ID! @unique
+        |   int: Int @unique
+        |   middles:[Middle!]!   @relation(name: "TopToMiddle", onDelete: CASCADE)
+        |}
+        |
+        |type Middle {
+        |   id: ID! @unique
+        |   int: Int! @unique
+        |   top: Top @relation(name: "TopToMiddle")
+        |   bottom: [Bottom!]! @relation(name: "MiddleToBottom", onDelete: CASCADE)
+        |}
+        |
+        |type Bottom {
+        |   id: ID! @unique
+        |   middle: Middle @relation(name: "MiddleToBottom")
+        |   int: Int!
+        |}
+      """
+    }
+    database.setup(project)
+
+    val top = server.query("""mutation {createTop(data:{int: 1}){int}}""", project)
+
+    def createMiddle(int: Int) = server.query(s"""mutation {createMiddle(data:{int: $int top: {connect:{int: 1}}}){int}}""", project)
+
+    for (int <- 1 to 200) {
+      createMiddle(int)
```

Comment:
Could we make this spec faster by optimizing the creation of those nodes? How about updating the parent node and then use nested mutations to create the middles?